### ISSUE TITLE
[Bug] Wrong References for the module variable

### ISSUE THREAD

**Issue by Peefy** [kcl-lang/kcl#1734]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

In the konfig repo https://github.com/kcl-lang/konfig/blob/main/models/kube/frontend/job.k

### 2. What did you expect to see? (Required)

The right reference list

### 3. What did you see instead (Required)

![image](https://github.com/user-attachments/assets/03fa2830-2b1d-4cab-b349-1763394995be)

### 4. What is your KCL components version? (Required)

v0.10.7

**Comment by dennybaa** [comment]
Possibly this one is related too, running `v0.10.7`:

```
import .providers  # module
resources = {
    #                         v
    #                  module
    "providers": providers.installProviders  # [Provider]
    #                              v
    #                        [Provider]
    "vault-config": providers.vaultConfig # error
}
```

```
35 |     "vault-config": providers.vaultConfig
   |                     ^ attribute 'vaultConfig' not found in '[Provider]'
```

Actually `resource.providers` definition at top of `resources` uses a locally set `providers`. From one side it's kinda expected and what it should be, but it used to work before. Please share what the actual future behavior should be?

**Comment by dennybaa** [comment]
Also, @Peefy  please note that 0.10.7 behaves strange with modules, they appear exposed similarly to variables. However this one is unexpected, since the scope for `import ...` identifiers should be better file only, but not anymore...

- sub/use_to_config.k (import ..config)
- sub/main.k (import .config)

```
.
├── config.k
├── kcl.mod
├── kcl.mod.lock
├── main.k
└── sub
    ├── config.k
    ├── main.k
    └── use_top_config.k
```
 
This clashes with 
```
   |
1 | import ..config
  | ^ the name 'config' is defined multiple times, 'config' must be defined only once
```

**Comment by dennybaa** [comment]
Hello, @Peefy I'd like to kindly ask about any plans for this? Also what bothers me a bit, is an unclear migration path from from 0.9.8 to 0.11.x.

1) First this issue which messes up variables with imports.
2) Second is the ordered file load behavior.  Let's say in `b.kcl` file has `var = true` and in `main.kcl` there is `output = var` this works fine on `0.9.x`. And in 0.11 seemed to like the order is broken (or it could be actually I couldn't just say for sure becouse of broken imports), or maybe it tries to behave like `cue` the file order doesn't matter.

Please could clarify this, since the scope for issues seems quite broad to me. So what the load behavior for KCL should actually be from now on?

Thank you!

**Comment by He1pa** [comment]
I want to confirm, does the problem you mentioned only occur in the IDE, or does it occur in the IDE and during the compilation process? Regarding the module-related behaves, there are some updates in the IDE. But I don't think this should affect the normal compilation process.

Regarding file dependencies and loading order, for entry files, they are in order, for example, `kcl run a.k b.k` or entries in kcl.mod, there is an order between a and b. For dependent files, such as import c in a.k. The correct order should be c a b

**Comment by dennybaa** [comment]
@He1pa yep, this unfortunately not the language sever issue only. It's compilation issue too.

**Comment by Peefy** [comment]
Closing as the symptom (module import names bleeding into local scope, attributes not found in array-of-schema lookups) has been touched by most of the recent resolver / scope-cache / evaluator fixes since this issue was opened:

  - #2097 (lazy scope order-independent globals)
  - #2075 (lazy scope cache on reassignment)
  - #2120 (eager/lazy mixin duplication)
  - #2128 (schema error location improvements)
  - #2132 (config literal type-check in schema defaults)

These directly address the two failure modes visible in the conversation thread (forward-reference type-checking on top-level lambdas; module-vs-variable scope confusion). Peefy also confirmed in this thread that the v0.10.8+ / v0.11.x line would carry the fix.

If you can still reproduce this on a current release (kcl 0.11.x) with a minimal -able repro, please reopen with: (a) a directory tree, (b) the exact .k files, and (c) the diagnostic or wrong-reference output. I'll dig into the specific resolver path.
